File: portfolio/src/main/java/dev/exit/asset/portfolio/CashFlowTransactionRepository.java
package dev.exit.asset.portfolio;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;

import dev.exit.asset.domain.CashFlowTransaction;

public interface CashFlowTransactionRepository extends JpaRepository<CashFlowTransaction, UUID> {

	List<CashFlowTransaction> findByOwnerIdOrderByTransactedOnDesc(String ownerId);

	List<CashFlowTransaction> findByOwnerIdAndTransactedOnBetween(String ownerId, LocalDate from, LocalDate to);
}

File: cashflow/src/main/kotlin/dev/exit/asset/cashflow/CashFlowTransactionRepository.kt
package dev.exit.asset.cashflow

import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate
import java.util.UUID

interface CashFlowTransactionRepository : JpaRepository<CashFlowTransaction, UUID> {

    fun findByOwnerIdOrderByTransactedOnDesc(ownerId: String): List<CashFlowTransaction>

    fun findByOwnerIdAndTransactedOnBetween(
        ownerId: String,
        from: LocalDate,
        to: LocalDate,
    ): List<CashFlowTransaction>
}
